ovh/types_dbaas_logs_role.go

```diff
@@ -0,0 +1,52 @@
+package ovh
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// DbaasLogsRoleCreateOpts defines the options for creating a role.
+type DbaasLogsRoleCreateOpts struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+}
+
+func (opts *DbaasLogsRoleCreateOpts) FromResource(d *schema.ResourceData) *DbaasLogsRoleCreateOpts {
+	opts.Name = d.Get("name").(string)
+	opts.Description = d.Get("description").(string)
+	return opts
+}
+
+// DbaasLogsRoleUpdateOpts defines the options for updating a role.
+type DbaasLogsRoleUpdateOpts struct {
```

Comment:
I guess this struct is useless since creation and update take the exact same parameters, it should be removed